### ISSUE TITLE
Game window starts on different and weird positions

### ISSUE THREAD

**Issue by lukycody** [Healix/Gw2Launcher#125]
Hey there. Using gw2launcher on a fresh & new gw2 installation setup all my account entries in the app.
And now I'm experiencing weird shenanigans with the window sizing and location:

##**Problem:** the game windows shows/launches on different positions for different accounts.
(basically: 1 Main Account and 19 other login-slaves with behave differently). this makes going through 20 accounts with "jumping windows all over the screen" a pain which definitely it triggers me. Before the fresh install every account started the same with its "black launching window" in the size/position correctly as set in the options.

### Case 1st account:
- Starts with an almost "fullscreen window".
![image](https://user-images.githubusercontent.com/63720164/150211579-c858a8bf-3a0f-49ed-8d70-7f468afc1e52.png)

- it then quickly "jumps" into a bigger window on the bottom right corner:
![image](https://user-images.githubusercontent.com/63720164/150211503-bf49668d-e568-4562-8180-397b2d53bb5c.png)

- Then with some delay it "snaps" into the actual set size and location (option "windows mode" checked with standard values of 854x, 480y, 853w, 480h)
![image](https://user-images.githubusercontent.com/63720164/150211684-6de799e2-097b-4ac6-bf2e-b5ea975aca6b.png)

### Case 2nd account and the rest (clones of 2nd with changed local.dat copy):
Some will directly go into the "final" set size/position. However, most of the time they also start with a "fullscreen window" . Then it turns into a medium size window on the top left corner and lastly snaps into the set small centered window (854x, 480y, 853w, 480h).
![image](https://user-images.githubusercontent.com/63720164/150213459-d4d37759-f801-43f3-8626-a69084fb5190.png)

## Another side-problem: when starting the instances by clicking on"play" after logging into an account, the fullsize starting window gets moved into the foreground (blocks other launcher windows) so you first have to select another gw2 instance on the taskbar to click "play".

How to change this, so that every started instance of the game launches with the same behavious?
It drives me crazy, send help :D

**Comment by Healix** [comment]
The initial size/position of GW2 is what GW2 has saved in Local.dat. Gw2Launcher applies your window settings after that, which is why you're seeing it jump. The full size window is because Local.dat was saved as full screen.

You'll need to launch each Local.dat using "Launch (normal)" to have GW2 save the initial size/position you're wanting.

**Comment by lukycody** [comment]
Thanks for your reply. I tried your suggestion and selected and launched the accounts separately via rightklick > selected > launch (normal) (only one at a time). However it did not work, still the same problem with the windows.

**Comment by Healix** [comment]
You need to launch (normal), load the game to character select, then exit normally (you can't force quit it or use the kill button in Gw2Launcher). Do 1 account, then launch and confirm it's saved.

If it's still not saving, launch (normal), open the in-game options > sound > toggle the last option "mute when GW2 is in the background". This is one of the settings saved exclusively in Local.dat, so if it reverts, you know GW2 isn't saving. If it did save, it should have also saved the window, but if it didn't, you may need to manually move the window or minimize and restore it to force an update.

If you have multiple screens, note that GW2 in DX11 doesn't support saving on secondary screens and will default to the top left corner of the primary screen.

**Comment by lukycody** [comment]
I tried it again exactly as you said but it didn't change the behaviour, that the big almost fullscreen-window pops up and then jumps to the smaller window. I also tested to toggle the "gw2 background option" and it indeed saved that change, but again, not the window behaviour. Could you specify how to force the change in that case? I minimized the gw2 instance aswell but it appears this didn't change anything.

Sorry, anything else that comes to your mind? This is really weird.

**Comment by Healix** [comment]
The big window is caused by DX9. To enable DX11 on all accounts, add -dx11 to the arguments under Settings > Guild Wars 2 > Launch options.

**Comment by lukycody** [comment]
Tried that aswell. The big window still pops up at the start but just way quicker, so it still kinda jumps into the specified location/size after that. I suppose you still have to start with "normal mode"? I tried that with the first account via manually choosing normal, and other accounts still got that medium sized window in between via just clicking the account to start it.
So yeah, it didn't really fix it :/

**Comment by Healix** [comment]
I can confirm the same problem - it's just a 1 frame flash for me with DX11. That's a flaw in how GW2 creates its window.

As for launching with DX9 or 11, there shouldn't be any noticeable difference if you were to _only_ load the launcher - everything is identical up to that point. Once you hit play to start the game is when DX11 begins. DX11 does require more resources, so it will affect the responsiveness of other clients.

**Comment by lukycody** [comment]
About dx11: so for multiboxing to grab daily rewards quickly it's recommended to stay with dx9 as it uses less resources?

About the other problem: Hm... alright thats a shame as with dx11 I still have the initial issues, that I described above (even after the "normal dx9 induced" big window flash I also get the weird "middle window position" until it jumps into the final window size/position).

Would it help to delete all the .dat files, create one new Local.dat by logging into the game once, then creating a new Local_XX.dat for each additional account login data and referencing them directly in the gw2launcher account objects?
What is the best/robust way to use the premade local.dats btw., if I remember correctly, I chose "copy from" and selected %appdata%/Guild Wars 2 where I created those before.

**Comment by Healix** [comment]
DX11 will eventually be forced, so you don't really have a choice. It's faster to load, but uses more resources to play in order to squeeze more FPS, which is worse for running multiple accounts. Keep in mind if you don't have the FPS limiter set, GW2 is going to take whatever it can get and DX11 will take more, so for best results, set it to the lowest option (30 FPS).

The middle window position means the position you're expecting wasn't saved in Local.dat. The window jumps like this: GW2 creates its window (this is the big window), loads Local.dat (this is the middle window), then Gw2Launcher applies your window settings (the final window). If you were to launch the account, then exit Gw2Launcher before clicking play, it should start with the big window and end with the middle window. Use Launch (normal) and load to character select - the current window position _should_ be saved at this point, but to make sure, move the window, then right click the account in Gw2Launcher > selected > apply windowed bounds, then load a character and exit (you may also want to toggle a sound option to confirm Local.dat was actually saved). Launch the account again to verify it.

For making a new Local.dat, the simplest option would be to copy the Local.dat from an account that is currently working the way you want it to, then use Launch (normal) to login and exit (to update the remembered login).

If you want to copy Local.dat files, the proper way is to edit the account > Local.dat > ... > copy from either an existing account or file. Alternatively, you can directly overwrite the #.dat files under %appdata%/Gw2Launcher/data, then you need to delete all the # folders within that folder, otherwise it may still use the old file (if using virtual mode).

**Comment by lukycody** [comment]
Thanks for the thorough explanation. Helped me to confirm my momentary understanding of how it works.
With this I will try again to overwrite the local.dat and make sure that it saves the new window position in the local.dat so that it's not as intrusive/annoying as it is now.

If I understand this correctly, only the one "original" Local.dat has to be forced to modify and you will always get 3 states of windows no matter what: firstly default "fullscreen windowed" window creation , secondly local.dat's last window setting, thirdly gw2launcher's forced window setting that gets applied lastly - which results in 2 consecutively fast changes of the window.

**Comment by Healix** [comment]
Yes, everyone experiences that initial big window - it even does it for full screen. Personally, I've never really noticed it as I've always had something else on screen that GW2 would launch behind.

If you don't want to see the window jump (between GW2's and Gw2Launcher's settings), disable the window size (or leave it blank) in Gw2Launcher. If you wanted all accounts to have identical windows, make 1 account the way you want it, then copy that Local.dat to all other accounts.
